src/xtgeo/grid3d/grid.py

```diff
@@ -706,66 +721,62 @@ def props(self):
         # Note, internally, the _props is a GridProperties instance, which is
         # a class that holds a list of properties.
 
-        prplist = None
         if isinstance(self._props, GridProperties):
-            prplist = self._props.props
+            return self._props.props
         elif isinstance(self._props, list):
             raise RuntimeError(
                 "self._props is a list, not a GridProperties " "instance"
             )
-        return prplist
+        return None
 
     @props.setter
-    def props(self, plist):
+    def props(self, plist: list[GridProperties]) -> None:
         if not isinstance(plist, list):
             raise ValueError("Input to props must be a list")
 
-        for litem in plist:
-            if litem.dimensions != self.dimensions:
+        for gridprop in plist:
+            if gridprop.dimensions != self.dimensions:
                 raise IndexError(
-                    f"Property NX NY NZ <{litem.name}> does not match grid!"
+                    f"Property NX NY NZ <{gridprop.name}> does not match grid!"
                 )
 
         self._props.props = plist  # self._props is a GridProperties instance
 
     @property
-    def propnames(self):
+    def propnames(self) -> Optional[list[str]]:
         """Returns a list of property names that are hooked to a grid."""
-        plist = None
-        if self._props is not None:
-            plist = self._props.names
-
-        return plist
+        return None if self._props is None else self._props.names
 
     @property
-    def roxgrid(self):
+    def roxgrid(self) -> Optional[Any]:
         """Get the Roxar native proj.grid_models[gname].get_grid() object."""
         return self._roxgrid
 
     @property
-    def roxindexer(self):
+    def roxindexer(self) -> Optional[Any]:
         """The Roxar native proj.grid_models[gname].get_grid().grid_indexer object."""
         return self._roxindexer
 
-    def generate_hash(self, hashmethod="md5"):
+    def generate_hash(
+        self,
+        hashmethod: Literal["md5", "sha256", "blake2b"] = "md5",
+    ) -> str:
         """Return a unique hash ID for current instance.
 
         See :meth:`~xtgeo.common.sys.generic_hash()` for documentation.
 
         .. versionadded:: 2.14
         """
         required = (
-            "ncol",
-            "nrow",
-            "nlay",
-            "coordsv",
-            "zcornsv",
-            "actnumsv",
+            "_ncol",
+            "_nrow",
+            "_nlay",
+            "_coordsv",
+            "_zcornsv",
+            "_actnumsv",
         )
 
-        gid = ""
-        for req in required:
-            gid += f"{getattr(self, '_' + req)}"
```

Comment:
Ive heard rummers that string concat is O(n*m) in python, small optimallzaion but i also feel that is more clear.